util/src/main/java/io/kubernetes/client/informer/cache/DeltaFIFO.java

```diff
@@ -443,7 +444,18 @@ private void syncKeyLocked(String key) {
    */
   private MutablePair<DeltaType, KubernetesObject> isDuplicate(
       MutablePair<DeltaType, KubernetesObject> d1, MutablePair<DeltaType, KubernetesObject> d2) {
-    return isDeletionDup(d1, d2);
+    MutablePair<DeltaType, KubernetesObject> deletionDelta = isDeletionDup(d1, d2);
+    if (deletionDelta != null) {
+      return deletionDelta;
+    }
+    if (d1.getLeft() != DeltaType.Deleted
```

Comment:
// TODO: remove this after the cause of memory leakage is confirmed
// Squashing deltas w/ the same resource version, note that is a temporary fix that eases memory intensity.